config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml

```diff
@@ -1297,6 +1297,7 @@ presubmits:
           curl -sSL https://kind.sigs.k8s.io/dl/latest/linux-amd64.tgz | tar xvfz - -C "${PATH%%:*}/" kind &&
           test/e2e/dra/kind-build-image.sh dra/node:latest &&
           kind create cluster --config test/e2e/dra/kind.yaml --image dra/node:latest &&
+          trap 'kind export logs "${ARTIFACTS}/kind"' EXIT &&
```

Comment:
trap _should_ work here bc of the way we pass things to `/bin/sh`, but it's definitely not my favourite or most intuitive way of handling this.

I've been out for a while, so will leave for some second eyes, but:

/approve

Reply:
The alternative is to extend the script at the end:
```
...
KUBERNETES_PROVIDER=local KUBECONFIG=${HOME}/.kube/config GINKGO_PARALLEL_NODES=8 E2E_REPORT_DIR=${ARTIFACTS} hack/ginkgo-e2e.sh -ginkgo.focus=DynamicResourceAllocation

ret=$?
kind export logs "${ARTIFACTS}/kind"
exit $ret
```

But then it would also attempt to export logs even when some of the earlier steps failed. More changes would be needed to avoid that, like breaking up the entire `<command> && <command> && ...` into individual steps.

This seemed like a shorter way of getting the desired effect. I can't deny that it isn't obvious, though. :sweat_smile: